src/test/cypress/integration/exam/ExamDateVerification.spec.ts

```diff
@@ -0,0 +1,159 @@
+                exam = response.body;
+                courseManagementRequests.registerStudentForExam(course, exam, artemis.users.getStudentOne());
+                cy.login(artemis.users.getStudentOne(), `/courses/${course.id}`);
+                cy.url().should('contain', `${course.id}`);
+                cy.contains('Exams').click();
+                cy.url().should('contain', '/exams');
+                cy.contains(examTitle).should('exist').click();
+                cy.url().should('contain', `/exams/${exam.id}`);
+            });
+        });
+
+        it('Student can start after start Date', () => {
+            let exerciseGroup: any;
+            let textExercise: any;
+            const student = artemis.users.getStudentOne();
+            const examContent = new CypressExamBuilder(course)
+                .title(examTitle)
+                .visibleDate(dayjs().subtract(3, 'days'))
+                .startDate(dayjs().subtract(2, 'days'))
+                .endDate(dayjs().add(3, 'days'))
+                .build();
+            courseManagementRequests.createExam(examContent).then((examResponse) => {
+                exam = examResponse.body;
+                courseManagementRequests.registerStudentForExam(course, exam, student);
+                courseManagementRequests.addExerciseGroupForExam(course, exam, 'group 1', true).then((groupResponse) => {
+                    exerciseGroup = groupResponse.body;
+                    courseManagementRequests.addTextExerciseToExam(exerciseGroup, 'Text exercise 1').then((exerciseResponse) => {
+                        textExercise = exerciseResponse.body;
+                        courseManagementRequests.generateMissingIndividualExams(course, exam);
+                        courseManagementRequests.prepareExerciseStartForExam(course, exam);
+                        cy.login(student, `/courses/${course.id}/exams`);
+                        cy.contains(exam.title).click();
+                        cy.url().should('contain', `/exams/${exam.id}`);
+                        cy.contains('Welcome to ' + exam.title).should('exist');
+                        cy.get('#confirmBox').click();
+                        artemis.users.getAccountInfo((account: any) => cy.get('#fullname').type(account.firstName + ' ' + account.lastName));
+                        cy.contains('Start').click();
+                        cy.contains('Exam Overview').should('exist');
+                        cy.intercept('PUT', `/api/exercises/${textExercise.id}/text-submissions`).as('savedSubmission');
+                        cy.contains('Text exercise 1').should('exist').click();
+                        cy.get('#text-editor-tab').type(
+                            'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
+                        );
+                        cy.contains('Save').click();
+                        cy.wait('@savedSubmission');
```

Comment:
I suggest moving the interception of the request directly before the UI interaction, which should trigger the request. Otherwise it seems like somewhere between the clicking on the text exercise and clicking the save button the request is sent. But actually the request is only sent after clicking on the save button.
```suggestion
                        cy.contains('Text exercise 1').should('exist').click();
                        cy.get('#text-editor-tab').type(
                            'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
                        );
                        cy.intercept('PUT', `/api/exercises/${textExercise.id}/text-submissions`).as('savedSubmission');
                        cy.contains('Save').click();
                        cy.wait('@savedSubmission');
```